### PR TITLE
Add sqlite-vss as a vector database

### PR DESCRIPTION
**Pull request by philippe2803**
This adds sqlite-vss as an option for a vector database. Contains the code and a few tests. Tests are passing and the library sqlite-vss is added as optional as explained in the contributing guidelines. I adjusted the code for lint/black/ and mypy. It looks that everything is currently passing.

Adding sqlite-vss was mentioned in this issue: https://github.com/langchain-ai/langchain/issues/1019. 
Also mentioned here in the sqlite-vss repo for the curious: https://github.com/asg017/sqlite-vss/issues/66

Maintainer tag: @baskaryan

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/sqlitevss.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        embedding: Embeddings,
+        db_file: str = "vss.db",
+    ):
+        """Initialize with sqlite client with vss extension."""
+        try:
+            import sqlite_vss  # noqa  # pylint: disable=unused-import
+        except ImportError:
+            raise ImportError(
+                "Could not import sqlite_vss python package. "
+                "Please install it with `pip install sqlite_vss`."
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/sqlitevss.py:46]: should it be sqlite-vss

File: libs/langchain/langchain/vectorstores/sqlitevss.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        """
+        cursor = self._connection.cursor()
+        cursor.execute(sql_query)
+        results = cursor.fetchall()
+
+        documents = []
+        for row in results:
+            metadata = json.loads(row["metadata"]) or {}
+            doc = Document(page_content=row["text"], metadata=metadata)
+            score = self._euclidean_relevance_score_fn(row["distance"])
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/sqlitevss.py:154]: i think we probably want to return raw distance here (the similarity_search_with_relevance_score method is where we'd want to normalize)

**Comment by baskaryan** [comment]
looks awesome, two quick comments!

File: libs/langchain/langchain/vectorstores/sqlitevss.py (hunk 1 above)
Comment (philippe2803) [reply libs/langchain/langchain/vectorstores/sqlitevss.py:46]: good catch!

File: libs/langchain/langchain/vectorstores/sqlitevss.py (hunk 2 above)
Comment (philippe2803) [reply libs/langchain/langchain/vectorstores/sqlitevss.py:154]: OK, I'll adjust accordingly 👍

**Comment by philippe2803** [comment]
@baskaryan thanks for your comments. I adjusted the code accordingly 👍

**Comment by baskaryan** [comment]
thanks @philippe2803!!

**Comment by baskaryan** [comment]
@philippe2803 if you have time would be great to also add a notebook demo-ing how to use it!

**Comment by philippe2803** [comment]
@baskaryan cool beans 👍 
I'll work on the notebook in the next few days, and will do an extra PR.

**Pull request merged**